apps/readest-app/src/app/reader/components/settings/MiscPanel.tsx

```diff
@@ -73,6 +74,12 @@ const MiscPanel: React.FC<{ bookKey: string }> = ({ bookKey }) => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [animated]);
 
+  useEffect(() => {
+    viewSettings!.disableClick = isDisableClick;
+    setViewSettings(bookKey, viewSettings!);
```

Comment:
Also need this as the `animated` logic: 
```
if (isFontLayoutSettingsGlobal) {
      settings.globalViewSettings. disableClick = disableClick;
      setSettings(settings);
}
```

Reply:
And `viewSettings` is assured to be valid above:
```
const viewSettings = getViewSettings(bookKey)!;
```
so no need to  re-ensure it like this `viewSettings!.disableClick = isDisableClick;` just
`viewSettings.disableClick = isDisableClick;` should be OK. It's better to use the `JavaScript and TypeScript Nightly` extension in VSCode. Generally it works fine to deduce when to use the non-null assertion(!).

Reply:
Thanks :)